### PR TITLE
Support for server header

### PR DESCRIPTION
**Pull request by geoffjohn11**

### CODE REVIEW

File: zio-http/src/main/scala/zio/http/model/headers/values/Server.scala [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+    /**
+     * No server name
+     */
+    object EmptyServerValue extends Server
+
+    def fromServer(server: Server): String =
+      server match {
+        case ServerValue(name) => name
+        case EmptyServerValue => ""
```
Comment (jdegoes) [inline zio-http/src/main/scala/zio/http/model/headers/values/Server.scala]: Is this empty, or is it invalid?

File: zio-http/src/main/scala/zio/http/model/headers/values/Server.scala [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+ * Server header value.
+ */
+sealed trait Server
+
+  object Server {
+    
+    /**
+     * A server value with a name
+     */
+    final case class ServerValue(name: String) extends Server
```
Comment (jdegoes) [inline zio-http/src/main/scala/zio/http/model/headers/values/Server.scala]: Maybe call this `Name`?

File: zio-http/src/main/scala/zio/http/model/headers/values/Server.scala (hunk 1 above)
Comment (geoffjohn11) [reply zio-http/src/main/scala/zio/http/model/headers/values/Server.scala]: In the MDN [docs](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Server) the value for server header is a name value with type String.  Some response look like "Apache/2.4.1 (Unix)" but twitter.com had a server header response value that was simply "tsa_b".  Therefore in the implementation any non-empty string value is accepted, and anything else is considered empty.  I'm not sure what a non-empty invalid server value would be, so an Invalid state wasn't implemented.  Any thoughts?

File: zio-http/src/main/scala/zio/http/model/headers/values/Server.scala (hunk 2 above)
Comment (geoffjohn11) [reply zio-http/src/main/scala/zio/http/model/headers/values/Server.scala]: I changed it to ServerName, is your preference for "Name"?

**Pull request merged**